### ISSUE TITLE
Fix accessibility issues with the minimal header and crisis line modal

### ISSUE THREAD

**Issue by rsmithadhoc** [department-of-veterans-affairs/vets-design-system-documentation#2220]
## Description

Must-fix issues from the [accessibility review](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2100) for this component.

## Details

- **Must:** Missing yellow focus outline on the VA logo.
    
    <img width="1095" alt="image" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/a4589f65-d484-4ea7-aca5-448f51855c90">

- **Must:** On the VA Logo link, there should be some text for screen readers. Adding an `aria-label` to the link should be sufficient.
    
    <img width="1791" alt="image" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/a08cca45-bf34-4902-8b75-2aa5c4dd22bc">

- **Must**: There were some issues with the crisis line modal, I don't see these issues on the main header in production.
    - Crisis line modal is not announced by screen readers. I think this needs `role="dialog"`.
    - You can tab outside of the modal to the VA logo link. This also means that focus is not trapped inside the modal, as it should be (see production's crisis modal).
        
        <img width="1791" alt="image" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/84fbc55d-9cb1-4485-9e7b-c4c79ce1b319">

## Tasks
- [ ] Fix the issues.
- [ ] Test with keyboard and screen reader to ensure the fixes corrected the issues.

## Acceptance Criteria
- [ ] Accessibility issues are corrected

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2220?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010

**Comment by jamigibbs** [comment]
Adding the `clickToClose` prop to va-modal may resolve the lack of focus trapping: https://design.va.gov/storybook/?path=/docs/components-va-modal--click-outside-to-close

**Comment by caw310** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2220?planning-poker) with Zenhub @rsmithadhoc

**Comment by rsmithadhoc** [comment]
@Andrew565 I saw a couple of issues:

- I'm still able to tab out of the modal to the VA logo. 
  
  <img width="1783" alt="image" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/dc0ad7cd-04a6-482c-8279-ffcfba00292c">

- It announces the modal now, but it reads out the entire content of the modal. I think the `aria-describedby="description"` is causing that and can be dropped. See screenshot below, it starts saying "If you are a veteran in crisis..." and keeps going.
  
  <img width="1783" alt="image" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/134a5c9b-b67e-4ad3-9ae9-f8b4caa7d604">

**Comment by Andrew565** [comment]
Will carry over due to a failing analytics-related test that I'm still working to resolve. CC: @caw310

**Comment by Andrew565** [comment]
The final "Must" - including CLM's Screen Reader issue and keyboard focus trapping for modals - will get addressed in #2255.
